ansel: init at unstable-2024-01-05

eclairevoyant opened the pull request:
## Description of changes

Adds ansel, a fork of darktable with different design goals.

Homepage: https://ansel.photos/

Fixes #269535.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

eclairevoyant committed:
ansel: init at unstable-2024-01-05

eclairevoyant commented:
I did test out some basic functionality, but @l0b0 please do give it a more thorough test just to make sure it's properly usable.

l0b0 commented:
LGTM, some observations:

- Quick build (3m31s on old desktop running other stuff at the same time) :+1: 
- Did not pick up my Darktable settings :-1: 
- Very low contrast between text and backgrounds :-1:  :-1: ![image](https://github.com/NixOS/nixpkgs/assets/168301/f4978672-52d8-4c58-8447-718276a2b0af)
- Esc goes from darkroom to lighttable :+1: 
- Folder view by default :+1: 
- Unsorted folder view, at least while importing :-1: 
- Loads image in lighttable much faster than Darktable :+1: 
- Imported 50k+ images with no trouble :+1:
- Standard-looking window menu :+1: :+1: ![image](https://github.com/NixOS/nixpkgs/assets/168301/d675990a-85d4-48d6-89db-da607bffe222)
- Space is not assigned in darkroom?? :-1:

thiagokokada commented on pkgs/top-level/all-packages.nix:
```diff
@@ -22460,6 +22460,17 @@ with pkgs;
 
   libavif = callPackage ../development/libraries/libavif { };
 
+  libavif_0_11 = libavif.overrideAttrs rec {
```
BTW, I don't like this approach of adding a named version only for one package (if there is more, this would be fine).

Can you overwrite `libavif` only for `ansel`?

eclairevoyant replied:
done, also added a comment explaining why it's needed.

eclairevoyant commented on pkgs/by-name/an/ansel/package.nix:
```diff
@@ -0,0 +1,156 @@
+{ libsepol
```
AIUI there's no enforced order as of #247424

eclairevoyant commented:
Addressed review comments to move the override into the ansel expression, enabled `strictDeps`, cleaned up some deps, updated to a more recent commit, and added `passthru.updateScript`. Please let me know if there's any other improvements to be made.

pbsds commented:
Result of `nixpkgs-review pr 269971` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ansel</li>
  </ul>
</details>

Seems to start fine. 
Please patch the shebangs in `share/ansel/tools/`

eclairevoyant commented:
thanks, fixed.

pbsds commented:
Result of `nixpkgs-review pr 269971` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ansel</li>
  </ul>
</details>

LGTM
